course6/toyapp-adam/util/clock.h

```diff
@@ -1,20 +1,23 @@
 #pragma once
 
-#include <SDL2/SDL.h>
+#include <chrono>
 
 namespace Util {
 
 class Clock
 {
+	using HiResClock = std::chrono::high_resolution_clock;
 public:
-	Clock();
+	Clock()
+	:	mStartTime(HiResClock::now())
+	{}
 
-	double GetTotalElapsedSeconds() const;
+	double GetTotalElapsedSeconds() const
+	{
+		return std::chrono::duration<double>(HiResClock::now() - mStartTime).count();
```

Comment:
I don't think that count will returns seconds. It will return "ticks", some abstract unit.
To get seconds, just divide the duration by 1 second:
```
using namespace std::literals;
return (HiResClock::now() - mStartTime) / 1s;
```
```

Reply:
That doesn't work, the returned value is always a whole number.